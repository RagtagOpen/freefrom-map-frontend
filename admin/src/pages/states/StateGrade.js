import React from 'react';
import PropTypes from 'prop-types';

function StateGrade(props) {

    const gradeClass = [
        'btn-success',
        'btn-success',
        'btn-warning',
        'btn-danger',
        'btn-danger'
    ]

    const gradeTitle = [
        'A', 'B', 'C', 'D', 'F'
    ]

    return (
        <div className={`btn ${ gradeClass[props.grade] }`}>
            { gradeTitle[props.grade] }
        </div>
    )
}

StateGrade.propTypes = {
    grade: PropTypes.number,
}

export default StateGrade;