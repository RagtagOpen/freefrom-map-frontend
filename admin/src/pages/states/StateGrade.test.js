import React from 'react';
import renderer from 'react-test-renderer';

import StateGrade from './StateGrade';

describe('State Grade Component', () => {
    let component;

    beforeEach(() => {
        component = renderer.create(<StateGrade grade={0} />);
    });

    it('should match snapshot', () => {
        expect(component.toJSON()).toMatchSnapshot();
    })
})
