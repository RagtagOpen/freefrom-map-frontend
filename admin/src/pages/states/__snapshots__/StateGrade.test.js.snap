// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`State Grade Component should match snapshot 1`] = `
<div
  className="btn btn-success"
>
  A
</div>
`;
